nodeadm/doc/examples.md

```diff
@@ -61,6 +61,38 @@ spec:
 
 The configuration objects will be merged in the order they appear in the MIME multi-part document, meaning the value in the lattermost configuration object will take precedence.
 
+---
+## Configuring node name (experimental)
+
+We introduced the ability to name nodes using their EC2 instance id instead of EC2 private DNS name for AL2023.
```

Comment:
I think we should talk about *why* this is a thing rather than just say it exists 😉 
```suggestion
When the `InstanceIdNodeName` feature gate is enabled, `nodeadm` will use the EC2 instance's ID (e.g. `i-abcdefg1234`) as the name of the `Node` object created by `kubelet`, instead of the EC2 instance's private DNS Name (e.g. `ip-192-168-1-1.ec2.internal`).

There are several benefits of doing this:
1. Your `Node` names are more meaningful in, for example, the output of `kubectl get nodes`.
2. The `Node` name, which is in the critical path of `kubelet` authentication, is non-volatile. While the private DNS name of an instance may change, its ID cannot.
3. The `ec2:DescribeInstances` permission can be removed from your node role's IAM policy; this is no longer necessary.
```